src/main/java/org/mule/extension/file/internal/source/PostActionGroup.java

```diff
@@ -52,14 +52,24 @@ public class PostActionGroup extends AbstractPostActionGroup {
   @Optional(defaultValue = "true")
   private boolean applyPostActionWhenFailed = true;
 
+  /**
+   * Whether the target file should be overwritten when the destination has the file with same name
+   */
+  @Parameter
+  @Optional(defaultValue = "false")
+  private boolean overwrite = false;
+
 
   public PostActionGroup() {}
 
-  public PostActionGroup(boolean autoDelete, String moveToDirectory, String renameTo, boolean applyPostActionWhenFailed) {
+
+  public PostActionGroup(boolean autoDelete, String moveToDirectory, String renameTo, boolean applyPostActionWhenFailed,
```

Comment:
Hi, Roger. The fact that you had to modify the test case is a probe that you are changing the API. 
A better option would be to create another constructor with five parameters and preserve the previous one with the default value for the new attribute.  
Doing so means you don't have to change any tests (you must add new ones, but without changing the previous ones).

Reply:
OK, got it.